presto-main/src/main/java/com/facebook/presto/operator/HttpPageBufferClient.java

```diff
@@ -151,32 +145,48 @@ public HttpPageBufferClient(
             HttpClient httpClient,
             DataSize maxResponseSize,
             Duration minErrorDuration,
+            Duration maxErrorDuration,
             URI location,
             ClientCallback clientCallback,
             BlockEncodingSerde blockEncodingSerde,
             ScheduledExecutorService executor)
     {
-        this(httpClient, maxResponseSize, minErrorDuration, location, clientCallback, blockEncodingSerde, executor, Stopwatch.createUnstarted());
+        this(httpClient, maxResponseSize, minErrorDuration, maxErrorDuration, location, clientCallback, blockEncodingSerde, executor, Ticker.systemTicker());
     }
 
     public HttpPageBufferClient(
             HttpClient httpClient,
             DataSize maxResponseSize,
             Duration minErrorDuration,
+            Duration maxErrorDuration,
             URI location,
             ClientCallback clientCallback,
             BlockEncodingSerde blockEncodingSerde,
             ScheduledExecutorService executor,
-            Stopwatch errorStopwatch)
+            Ticker ticker)
     {
         this.httpClient = requireNonNull(httpClient, "httpClient is null");
         this.maxResponseSize = requireNonNull(maxResponseSize, "maxResponseSize is null");
-        this.minErrorDuration = requireNonNull(minErrorDuration, "minErrorDuration is null");
         this.location = requireNonNull(location, "location is null");
         this.clientCallback = requireNonNull(clientCallback, "clientCallback is null");
         this.blockEncodingSerde = requireNonNull(blockEncodingSerde, "blockEncodingManager is null");
         this.executor = requireNonNull(executor, "executor is null");
-        this.errorStopwatch = requireNonNull(errorStopwatch, "errorStopwatch is null").reset();
+        requireNonNull(minErrorDuration, "minErrorDuration is null");
+        requireNonNull(maxErrorDuration, "maxErrorDuration is null");
+        requireNonNull(ticker, "ticker is null");
+        this.backoff = new Backoff(
+                minErrorDuration,
+                maxErrorDuration,
+                ticker,
+                new Duration(0, MILLISECONDS),
```

Comment:
This is how the code works today, which is why I used it. I agree it seems aggressive though. I'll change it to the default one